packages/x-date-pickers/src/timeViewRenderers/timeViewRenderers.tsx

```diff
@@ -50,13 +50,13 @@ export const renderTimeViewClock = <TDate extends unknown>({
   autoFocus,
   showViewSwitcher,
   disableIgnoringDatePartForTimeValidation,
-}: TimeViewRendererProps<TimeView, TimeClockProps<TDate>>) => (
+}: TimeViewRendererProps<TimeViewWithMeridiem, TimeClockProps<TDate, TimeViewWithMeridiem>>) => (
   <TimeClock<TDate>
-    view={view}
+    view={view as TimeView}
```

Comment:
A bit hacky, I know. 🙈 
But do you know any better way how to support all the time renderers on `DesktopTimePicker`, which expects `TimeViewWithMeridiem` capable renderers. 🤷